library/system/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,301 @@
+    def_delegator :@service, :sub_state, :substate
+
+    class << self
+      # Find a service
+      #
+      # @param name [String] Service name
+      # @return [SystemService,nil] System service or nil when not found
+      def find(name)
+        new(Yast::SystemdService.find(name))
+      end
+
+      # Finds service names
+      #
+      # This method finds a set of system services. Currently it is just a wrapper around
+      # SystemdService.find_many.
+      #
+      # @param names [Array<String>] Service names to find
+      # @return [Array<SystemService>] Found system services
+      def find_many(names)
+        Yast::SystemdService.find_many(names).compact.map { |s| new(s) }
+      end
+    end
+
+    # @param service [Yast::SystemdServiceClass::Service]
+    def initialize(service)
+      @service = service
+      @changes = {}
+    end
+
+    # Returns the start mode
+    #
+    # See {#start_modes} to find out the supported modes for a given service (usually :on_boot,
+    # :manual and, in some cases, :on_demand).
+    #
+    # When the service (:on_boot) and the socket (:on_demand) are enabled, the start mode is translated
+    # to :on_boot.
+    #
+    # @return [Symbol] Start mode (:on_boot, :on_demand, :manual)
+    def start_mode
+      new_value_for(:start_mode) || current_start_mode
+    end
+
+    # Sets the service start mode
+    #
+    # See {#start_modes} to find out the supported modes for a given service (usually :on_boot,
+    # :manual and, in some cases, :on_demand). The given value will be applied after calling #save.
+    #
+    # @see #start_modes
+    # @raise ArgumentError when mode is not valid
+    def start_mode=(mode)
+      if !start_modes.include?(mode)
+        raise ArgumentError, "Invalid start mode: '#{mode}' for service '#{service.name}'"
+      end
+
+      if mode == current_start_mode
+        unregister_change(:start_mode)
+      else
+        register_change(:start_mode, mode)
+      end
+    end
+
+    # Determine whether the service will be active after calling #save
+    #
+    # @return [Boolean] true if the service must be active; false otherwise
+    def active?
+      return new_value_for(:active) if changed_value?(:active)
+      service.active?
+    end
+
+    # Sets the service to be started after calling #save
+    #
+    # @see #active=
+    def start
+      self.active = true
+    end
+
+    # Sets the service to be stopped after calling #save
+    #
+    # @see #active=
+    def stop
+      self.active = false
+    end
+
+    # Toggles the service status
+    #
+    # @see #active=
+    def toggle
+      self.active = !active?
+    end
+
+    # Saves changes to the underlying system
+    #
+    # @param set_status [Boolean] Do not change service status. Useful when running on 1st stage.
+    def save(ignore_status: false)
+      save_start_mode
+      set_current_status unless ignore_status
+      reload
+    end
+
+    # Reverts stored changes
+    def reload
```

Comment:
I really really dislike this API name as it is confusing as service also has method reload. Why not simply call it read?

Reply:
or reset?